### PR TITLE
Stub out the rust crypto implementation for browserify

### PR DESCRIPTION
**Pull request by richvdh**
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/3182.

Even this took me an annoyingly long time to figure out. We *really* need to figure out if browserify is important to us.

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by richvdh** [comment]
gah, this breaks webpack as well

File: package.json [hunk 1]
```diff
@@ -33,7 +33,10 @@
         "matrix-org"
     ],
     "main": "./src/index.ts",
-    "browser": "./src/browser-index.ts",
+    "browser": {
```
Comment (t3chguy) [inline package.json:36]: This field is managed by release.sh so that would need updating to not revert your fix

Comment (richvdh) [reply package.json]: gordon bennett. Why are we still using JSON as a config file format in 2023?

Comment (richvdh) [reply package.json]: thanks for the heads-up, anyway. My more recent attempt doesn't change this, so it's moot.

File: package.json [hunk 2]
```diff
@@ -147,5 +148,25 @@
...
+            "browserify-swap",
+            [
+                "babelify",
+                {
+                    "sourceMaps": "inline",
+                    "presets": [
+                        "@babel/preset-env",
+                        "@babel/preset-typescript"
+                    ]
+                }
```
Comment (richvdh) [inline package.json:163]: moving this here wasn't a requirement, but the commandline was getting increasingly unwieldy.

File: src/rust-crypto/browserify-index.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
```
Comment (t3chguy) [inline src/rust-crypto/browserify-index.ts:2]:
```suggestion
Copyright 2023 The Matrix.org Foundation C.I.C.
```

**Review by t3chguy — APPROVED** [review]
Looks great to me, thanks!!

**Comment by t3chguy** [comment]
(Feel free to bypass sonar here)

**Pull request merged**